# Exploración de los Datos

**Autor:** Isabel Castrillón Acosta

**Fecha:** 26-06-2025

**Descripción:** Este notebook realiza una exploración inicial de los datos de la prueba técnica 2. Incluye posibles limpez, análisis de datos faltantes y visualizaciones básicas.

# 1. Librerias

In [1]:
import pandas as pd


# 2. Cargar los datos


In [2]:
df = pd.read_excel('../Datos/sentencias_pasadas.xlsx')
df

#  Relevancia Providencia  Tipo Fecha Sentencia  \
0      1    966965.0    T-185/22   NaN      2022-05-31   
1      3    963168.0    T-356/21   NaN      2021-10-15   
2      5    956201.0    T-351/22   NaN      2022-10-07   
3      6    955889.0    T-246/21   NaN      2021-07-29   
4      7    955787.0   T-245A/22   NaN      2022-07-01   
..   ...         ...         ...   ...             ...   
324  487    106681.0   SU.016/21   NaN      2021-01-21   
325  489    100241.0   SU.257/21   NaN      2021-08-05   
326  491     99781.0    C-489/23   NaN      2023-11-16   
327  493     84998.0    C-293/20   NaN      2020-08-05   
328  495     81298.0    C-294/21   NaN      2021-09-02   

                                        Tema - subtema  \
0                                                  NaN   
1    ACCIÓN DE TUTELA PARA PROTEGER EL DERECHO A LA...   
2    ACCIÓN DE TUTELA PARA PROTEGER EL DERECHO A LA...   
3    ACCION DE TUTELA PARA PROTEGER EL DERECHO A LA...   
4    ACCION DE TUTELA-Inexistencia de hecho superad...   
..                                                 ...   
324  ABUSO DEL DERECHO-Elementos que lo configuran ...   
325  ACCION DE TUTELA CONTRA PROVIDENCIAS JUDICIALE...   
326                                                NaN   
327                                                NaN   
328  NIÑOS, NIÑAS Y ADOLESCENTES COMO SUJETOS DE ES...   

                                              resuelve  \
0    en nombre del pueblo y por mandato de la Const...   
1    en nombre del pueblo y por mandato de la Const...   
2    administrando justicia en nombre del Pueblo, y...   
3    en nombre del pueblo y por mandato de la Const...   
4    en nombre del pueblo y por mandato de la Const...   
..                                                 ...   
324  en nombre del pueblo y por mandato de la Const...   
325  en nombre del pueblo y por mandato de la Const...   
326  RESUELVE ÚNICO. Declarar INEXEQUIBLE el parágr...   
327  administrando justicia en nombre del Pueblo, y...   
328  RESUELVE: PRIMERO.- Declarar la INEXEQUIBILIDA...   

                                              sintesis  
0    En este caso se formula la acción de tutela en...  
1    El peticionario considera que los accionantes ...  
2    El periodista accionante acusa al abogado acci...  
3    Se presenta la acción de tutela en contra de u...  
4    El accionante, actuando en representación de s...  
..                                                 ...  
324  En este caso, el accionante y 56 personas más,...  
325  La acción de tutela se interpuso en contra de ...  
326  Demanda de inconstitucionalidad contra el artí...  
327  Revisión de constitucionalidad del Decreto Leg...  
328  Demandas de inconstitucionalidad contra el Act...  

[329 rows x 8 columns]

# 3. Exploración inicial

In [3]:
df.columns

Index(['#', 'Relevancia', 'Providencia', 'Tipo', 'Fecha Sentencia',
       'Tema - subtema', 'resuelve', 'sintesis'],
      dtype='object')

In [4]:
df.shape

(329, 8)

In [5]:
df.isna().sum()

#                    0
Relevancia           0
Providencia          0
Tipo               329
Fecha Sentencia      0
Tema - subtema      92
resuelve             0
sintesis             0
dtype: int64

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 329 entries, 0 to 328
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   #                329 non-null    int64         
 1   Relevancia       329 non-null    float64       
 2   Providencia      329 non-null    object        
 3   Tipo             0 non-null      float64       
 4   Fecha Sentencia  329 non-null    datetime64[ns]
 5   Tema - subtema   237 non-null    object        
 6   resuelve         329 non-null    object        
 7   sintesis         329 non-null    object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(4)
memory usage: 20.7+ KB


In [7]:
print("Fecha min:",df['Fecha Sentencia'].min())
print("Fecha max:",df['Fecha Sentencia'].max())


Fecha min: 2019-02-01 00:00:00
Fecha max: 2024-11-13 00:00:00


# 4. Observaciones

- DF tiene 8 columnas y 329 filas.

- Columna 'Tipo' esta completamente vacía, por lo que no aporta información y se puede eliminar.

- la majoria de las columnas contienen objetos, lo que indica que pueden ser cadenas de texto. Pero también tienen integer, float y datetime.

- Las sentencias se hicieron entre 01-02-2019 y 13-11-2024.

